operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/ReconciliationDispatcher.java

```diff
@@ -109,27 +110,76 @@ private boolean shouldNotDispatchToDelete(R resource) {
       updateCustomResourceWithFinalizer(resource);
       return PostExecutionControl.onlyFinalizerAdded();
     } else {
-      log.debug(
-          "Executing createOrUpdate for resource {} with version: {} with execution scope: {}",
-          getName(resource),
-          getVersion(resource),
-          executionScope);
+      try {
+        var resourceForExecution =
+            cloneResourceIfErrorStatusHandlerIfNeeded(resource, context);
+        return createOrUpdateExecution(executionScope, resourceForExecution, context);
+      } catch (RuntimeException e) {
+        handleLastAttemptErrorStatusHandler(resource, context, e);
+        throw e;
+      }
+    }
+  }
+
+  /**
+   * Resource make sense only to clone for the ErrorStatusHandler. Otherwise, this operation can be
+   * skipped since it can be memory and time-consuming. However, it needs to be cloned since it's
+   * common that the custom resource is changed during an execution, and it's much cleaner to have
+   * to original resource in place for status update.
+   */
+  private R cloneResourceIfErrorStatusHandlerIfNeeded(R resource, Context context) {
```

Comment:
The more I think about it, the less I'm sure we need to clone before updating the status. Presumably, if the resource has changed, the reconciliation will happen again and this update should be cancelled, anyway?

Reply:
as mentioned in the docs, it is for the case when the reconciliation executed but at some point an exception thrown, meannwhile in the logic of `reconcile` the status could be updated on the CR. I don't think we should use that, rather the original version of the resource. Since also info could be deleted, it's probably better to work with the fresh copy.

Reply:
Not convinced but I'll trust your judgement on this. :)